src/framework/components/rigid-body/component.js

```diff
@@ -939,6 +939,9 @@ class RigidBodyComponent extends Component {
                 const q = ammoTransform.getRotation();
                 this.entity.setPosition(p.x(), p.y(), p.z());
                 this.entity.setRotation(q.x(), q.y(), q.z(), q.w());
+
+                this._getEntityTransform(ammoTransform);
+                motionState.setWorldTransform(ammoTransform);
```

Comment:
This looks odd to me as the ammoTransform should be the same value as what we got on line 936.

This change implies that setting the position and rotation of the entity is wrong or that _getEntityTransform is giving a different value to what we started with ammoTransform.

Reply:
It did to me as well:
![image](https://user-images.githubusercontent.com/1382022/198557497-83bd2f80-121c-4668-993d-c5724f0855b8.png)
That also fixes it, which I think points the issue to inside of `setRotation()`.
Because `getRotation()` overwrites the rotation with the world transform.
https://github.com/playcanvas/engine/blob/bf31671060633d8e26dff833ded278fcf14d5315/src/scene/graph-node.js#L853

Reply:
I haven't looked in details, but this looks odd as well. The motion state of a dynamic body is and should be driven by Ammo under normal update cycle. The issue must be elsewhere.

Reply:
Not sure if it's intended, but shouldn't this refer to `this`?
https://github.com/playcanvas/engine/blob/bf31671060633d8e26dff833ded278fcf14d5315/src/scene/graph-node.js#L1120

Reply:
It is intended, as it is using a private variable:
https://github.com/playcanvas/engine/blob/bf31671060633d8e26dff833ded278fcf14d5315/src/scene/graph-node.js#L20

Reply:
Yes - but getRotation doesn't read from that global rotation.

Reply:
Now I see the issue, I'm referring to `entity.rotation` and not `entity.getRotation`, but its weird as `.rotation` seems to a public property that does not accurately reads the rotation.

Reply:
```
        /**
         * @type {Quat}
         * @private
         */
        this.rotation = new Quat();
```

It shouldn't show up on code predication as it's tagged by JS Docs as private